ext/uri/uri_parser_rfc3986.c

```diff
@@ -214,6 +279,26 @@ ZEND_ATTRIBUTE_NONNULL static zend_result php_uri_parser_rfc3986_port_read(const
 	return SUCCESS;
 }
 
+static zend_result php_uri_parser_rfc3986_port_write(struct uri_internal_t *internal_uri, zval *value, zval *errors)
+{
+	UriUriA *uriparser_uri = get_uri_for_writing(internal_uri);
+	int result;
+
+	if (Z_TYPE_P(value) == IS_NULL) {
+		result = uriSetPortTextMmA(uriparser_uri, NULL, NULL, mm);
+	} else {
+		ZVAL_STR(value, zend_long_to_str(Z_LVAL_P(value)));
+		result = uriSetPortTextMmA(uriparser_uri, Z_STRVAL_P(value), Z_STRVAL_P(value) + Z_STRLEN_P(value), mm);
+	}
+
+	if (result != URI_SUCCESS) {
+		zend_throw_exception(uri_invalid_uri_exception_ce, "The specified port is malformed", 0);
```

Comment:
There is `URI_ERROR_SETPORT_HOST_NOT_SET` as a port-specific error code, that we should handle to output a better message:

```suggestion
		switch (result) {
		case URI_SUCCESS:
			return SUCCESS;
		case URI_ERROR_SETPORT_HOST_NOT_SET:
			zend_throw_exception(uri_invalid_uri_exception_ce, "Cannot set a port without having a host", 0);
			return FAILURE;
		default:	
			zend_throw_exception(uri_invalid_uri_exception_ce, "The specified port is malformed", 0);
			return FAILURE;
		}
```

The same applies for the other error codes that have a special meaning.